core/src/main/java/org/jdbi/v3/core/statement/SqlLogger.java

```diff
@@ -27,15 +27,27 @@
     SqlLogger NOP_SQL_LOGGER = new SqlLogger() {
     };
 
+    /**
+     * Will be called before a query is executed. Note that no timing information will be available in the {@link StatementContext} yet.
+     */
     default void logBeforeExecution(StatementContext context) {
     }
 
+    /**
+     * Will be called after a query has been executed. The {@link StatementContext} will contain a start {@link StatementContext#getExecutionMoment} and end {@link StatementContext#getCompletionMoment} time.
+     */
     default void logAfterExecution(StatementContext context) {
     }
 
+    /**
+     * Will be called after a query has failed. The {@link StatementContext} will contain a start {@link StatementContext#getExecutionMoment} and failure {@link StatementContext#getExceptionMoment} time.
+     */
     default void logException(StatementContext context, SQLException ex) {
     }
 
+    /**
+     * It's probably best not to override this method. It's only here due to architectural trade-off choices.
+     */
```

Comment:
I kinda still want to make this method a static utility method instead of a default method.

Reply:
Where would you put it? Can't go on an interface.

Reply:
https://github.com/jdbi/jdbi/pull/1144

Reply:
Perfect. Thanks